cheroot/connections.py

```diff
@@ -155,15 +158,29 @@ def get_conn(self, server_socket):
 
             # Wait for the next tick to occur.
             return None
-
-        try:
-            # See if we have a new connection coming in.
-            rlist.remove(ss_fileno)
-        except ValueError:
-            # No new connection, but reuse existing socket.
-            conn = socket_dict[rlist.pop()]
-        else:
-            conn = server_socket
+        finally:
+            for sock in socket_dict:
+                self._selector.unregister(sock)
+
+        # See if we have a new connection coming in.
+        new_connection = False
+        rlist = []
+        for key, event in key_events:
+            if key.fd == ss_fileno:
+                # If we have a new connection, reuse the server socket
+                new_connection = True
+                conn = server_socket
+            else:
+                # Make a list of all the other sockets ready to read
+                rlist.append(key.fd)
```

Comment:
paired with https://github.com/cherrypy/cheroot/pull/301/files#r456684464 this could be
```suggestion
        try:
            key_events.pop(ss_fileno)
        except KeyError:
                new_connection = False
        else:
                new_connection = True
                # If we have a new connection, reuse the server socket
                conn = server_socket

        # Make a list of all the other sockets ready to read
        rlist = list(key_events.keys())
```